### DISCUSSION TITLE
Import a script from the embed element

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#1892] [Ideas]
Hello,

I need to use leaflet and I would like to know if I can import a script using the embed element?

Thank you

**Comment by mjashanks** [comment]
This should be possible. You can use bindings to feed data into your script. You might will have a problem if you need to get data back out of the Embed component - e.g. to save to your DB.

**Reply by oswaldB** [reply]
Hello,

It seems working only for the css script:
![image](https://user-images.githubusercontent.com/3023526/124102447-484e4b80-da58-11eb-84d0-5f79b633345b.png)
and it works:
![image](https://user-images.githubusercontent.com/3023526/124102536-5d2adf00-da58-11eb-87a9-32c62faa7bf9.png)

but not for the js script:
![image](https://user-images.githubusercontent.com/3023526/124102625-716edc00-da58-11eb-8310-cab019592f3f.png)
and it doesn't work:
![image](https://user-images.githubusercontent.com/3023526/124102707-821f5200-da58-11eb-95ba-52e55e845bb8.png)

**Reply by shogunpurple** [reply]
@oswaldB our current implementation does not allow for the inclusion of a script tag. This is due to security/XSS concerns. We may allow for this functionality in future, but it is not on our immediate roadmap.

**Reply by mintonan** [reply]
Hi @shogunpurple I have looked around but haven't been able to find anything, has the script tag been implemented? I am trying to open an external app for my users and am not sure if I am doing something wrong or if there is no implementation of this

**Reply by shogunpurple** [reply]
Hi @mintonan it has not - you could use an embed component to do this or a link.

**Comment by oswaldB** [comment]
I understand. So there is no way for me right now to add leaflet - instead
if I change the source code?

On Thu, 1 Jul 2021 at 10:54, Martin McKeaveney ***@***.***>
wrote:

> @oswaldB <https://github.com/oswaldB> our current implementation does not
> allow for the inclusion of a script tag. This is due to security/XSS
> concerns. We may allow for this functionality in future, but it is not on
> our immediate roadmap.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/1892#discussioncomment-948759>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXCFJVMMDAH2NALI5FDIETTVQ3L5ANCNFSM47T5RRXA>
> .
>


-- 
Oswald BERNARD
00 44 7861004200

**Reply by mkell85** [reply]
@oswaldB I was able to load leaflet using the following configuration 
![leaflet](https://user-images.githubusercontent.com/512391/126267943-ffd86a20-94b3-41be-afeb-e12a6507ec70.PNG)

**Reply by oswaldB** [reply]
Yes you can embed it. But you cannot link the data in your table to the map. Because to do that, you need to have the right to import scripts...

Thank you

**Reply by dncpax** [reply]
Hey there. I'm also working with maps (openlayers). I embed a whole page with scripts just fine, and pass arguments by binding the url querystring to budibase fields. For instance I can zoom to an extent defined in a bb text input.

What is sorely missed is the ability to pass information from the iframe to bb!! If only we could use window.postMessage()... that would be enough... we could then use the new javascript transformers to massage the incoming data and set new values into bb fields or variables.

**Reply by lancedouglas1** [reply]
@dncpax is it possible to use a flux-ish approach in your embedded page that the interaction is passed to an api endpoint and the bb application is listening to the datasource where those updates were pushed.

Also, would love to see some details on how you're currently embedding with binding from bb to embedded page.

**Reply by dncpax** [reply]
hey @lancedouglas1 that seems possible but so convoluted...
The way I did was to just have the query params of the url built with bindings, so effectively reloading the page... not smooth but workable.
I still think that having postMessage and receivedMessage would be the best integration and the simplest mechanism. Also standard.

**Comment by igbojionu** [comment]
HI iis it possible to add custom JS to budibase or include [CDN]  i would want to have an app that can generate barcodes

**Reply by mjashanks** [reply]
The only way is via the `Embed` component, as mentioned above. Sometime (hopefully this year) we will allow you to build your own custom components - which could include scripts.

**Comment by Bytelegion** [comment]
Hi  @mjashanks , 

Can we import script tag now , we need to add some chat widgets to our screen if possible ?

**Reply by lvanoverberghe** [reply]
I need to integrate a chat too, with a script tag, but doing it with an embed object, I am getting Unexpected Identifier when running the app.

**Comment by c-nv-s** [comment]
also searching for a way to embed scripts over here.
Ideally would like to pass data from/to budibase with them too.

**Reply by SufiSR** [reply]
I managed to embed scripts. Please look at this tutorial:

https://github.com/Budibase/budibase/discussions/12143
